sd_dynamic_prompts/dynamic_prompting.py

```diff
@@ -520,6 +522,20 @@ def process(
             negative_prompts=all_negative_prompts,
         )
 
+        if opts.dp_write_raw_template:
+            # if original_prompt / original_negative_prompt will not be auto quoted by webui and will be effected by stripped, we need to quote it ourselves
+            if original_prompt and original_prompt != original_prompt.strip() and original_prompt == generation_parameters_copypaste.quote(original_prompt):
+                original_prompt = json.dumps(original_prompt, ensure_ascii=False)
+            if original_negative_prompt and original_negative_prompt != original_negative_prompt.strip() and original_negative_prompt == generation_parameters_copypaste.quote(original_negative_prompt):
+                original_negative_prompt = json.dumps(original_negative_prompt, ensure_ascii=False)
+
+            p.extra_generation_params.update(
```

Comment:
If updating `p.extra_generation_params` is the correct orthodox way of saving PNG info, I'm all for it, and please eradicate the current `pnginfo_saver` altogether then 😄

Reply:
> eradicate the current pnginfo_saver altogether then

I wish I could, `strip_template_info` need to stay forward backwards compatibility with existing images with the invalid pnginfo